openedx/core/djangolib/blockstore_cache.py

```diff
@@ -143,19 +144,18 @@ def get_bundle_version_number(bundle_uuid, draft_name=None):
     return version
 
 
+@lru_cache(maxsize=200)
 def get_bundle_version_files_cached(bundle_uuid, bundle_version):
     """
     Get the files in the specified BundleVersion. Since BundleVersions are
-    immutable, this should be cached as aggressively as possible.
+    immutable, this should be cached as aggressively as possible (ideally
+    it would be infinitely but we don't want to use up all available memory).
+    So this method uses lru_cache() to cache results in process-local memory.
+
+    (Note: This can't use BundleCache because BundleCache only associates data
+    with the most recent bundleversion, not a specified bundleversion)
     """
-    # Use the blockstore django cache directly; this can't use BundleCache because BundleCache only associates data
-    # with the most recent bundleversion, not a specified bundleversion
-    cache_key = 'bundle_version_files:{}:{}'.format(bundle_uuid, bundle_version)
-    result = cache.get(cache_key)
-    if result is None:
-        result = blockstore_api.get_bundle_version_files(bundle_uuid, bundle_version)
-        cache.set(cache_key, result, timeout=None)  # Cache forever since bundle versions are immutable
```

Comment:
I believe now that this was the problem: I assumed that bundleversions are immutable so started caching them in memcached with no timeout, but forgot that these data structures can encode signed S3 URLs, and it seems we are using the default `AWS_QUERYSTRING_EXPIRE` defaults to 1 hour - https://django-storages.readthedocs.io/en/latest/backends/amazon-S3.html . Quick fix for now is to only cache these for up to 1 hour, better long term solution is to change the URLs to be redirects (managed by blockstore) so that they can be cached forever. That should also improve performance since we won't need to generate signed URLs for all files in advance.